Sources/Network/HTTPResourceEndpoint.swift

```diff
@@ -59,8 +60,13 @@ public extension HTTPResourceEndpoint {
 
         urlRequest.httpMethod = method.rawValue
         urlRequest.allHTTPHeaderFields = headers
-        urlRequest.httpBody = body
+        urlRequest.httpBody = try makeBody()
 
         return urlRequest
     }
 }
+
+extension HTTPResourceEndpoint where Self: Encodable {
+
+    func makeBody() throws -> Data? { try JSONEncoder().encode(self) }
```

Comment:
I am bit confused what's the intent here.
Should a model implement `HTTPResourceEndpoint`?

Reply:
In short, yes. You can have a model which is something like `PotatoRequest: Encodable`, which you then use to build your request's JSON body. 

This also works for more complex cases, like:

```swift
enum MyAPIEndpoint: HTTPResourceEndpoint {
    case .banana(BananaRequest)
    case .potato(PotatoRequest)
}

extension BananaRequest: Encodable { ... } 
extension PotatoRequest: Encodable { ... }

extension MyAPIEndpoint: Encodable {
 
    func encode(to encoder: Encoder) throws {
        switch self {
        } 
    }

```

Reply:
On a second thought, this extension is also assuming that if the endpoint is `Encodable` you want JSON encoding on your request's body... 😅 That being said, perhaps it's safer to not make it the "default" behavior. 🤔

Reply:
I've since removed this default implementation because it was making too much assumptions 😇